crates/blockchain/dev/block_producer.rs

```diff
@@ -1,7 +1,7 @@
 use crate::utils::engine_client::{errors::EngineClientError, EngineClient};
 use bytes::Bytes;
 use ethereum_types::{Address, H256};
-use ethrex_rpc::types::fork_choice::{ForkChoiceState, PayloadAttributesV3};
+use ethrex_rpc::types::fork_choice::{ForkChoiceState, PayloadAttributes};
```

Comment:
This is outside of the RPC crate. This is what I talk about when I say that the abstraction is leaking. From outside, you want to deal with specific versions of payload attributes. The specific version is NOT an implementation detail, it is part of the spec (see https://ethereum.github.io/execution-apis/api-documentation/), consumers need to know about it.

So here you are change something that was explicit (V3) to something implicit. I don't know which version of the payload this is.